client/my-sites/email/inbox/new-mailbox-upsell/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import { recordInboxNewMailboxUpsellClickEvent } from 'calypso/my-sites/email/email-management/home/utils';
+import { INBOX_SOURCE } from 'calypso/my-sites/email/inbox/constants';
+import { emailManagement, emailManagementEdit } from 'calypso/my-sites/email/paths';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+const NewMailboxUpsell = ( { domains }: { domains: Array< ResponseDomain > } ) => {
```

Comment:
Nit, but the notation `ResponseDomain[]` is probably more common compared to `Array< ResponseDomain >` (at least in Calypso). So long as we don't have any specific reason for using `Array<>` (e.g. we are defining multiple allowed types like `Array< string | null >`), I think it makes sense to go with `ResponseDomain[]`.